### ISSUE TITLE
Disabling Dragon Roosts also disables Dragon Caves in the config [Latest version]

### ISSUE THREAD

**Issue by Fogdile** [Alex-the-666/Ice_and_Fire#3736]
I'm trying to create a world where dragons don't live on the surface, only in caves. 
So I turned dragon roosts to false, and kept dragon caves to true in the configs. However when I do that, dragons stop spawning all together. There are no caves even thought I've kept dragon caves on. I've searched thousands of blocks in spectator mode + used the eye of dragon mod to help me locate the caves. The Eye of dragon stays unresponsive because there are no dragons it can search for.
If I turn dragon roosts back on, dragon caves start spawning again. I've set set dragon roost chance to be 1/10000 and dragon caves will spawn normally, it's only a problem if I turn dragon roost spawn to false.
in a nutshell: setting dragon roosts = false also sets dragon caves to false when it shouldn't

1. Turn dragon roosts to false but keep dragon caves on true in the config file.
2. Load into a world and try to find a dragon cave.
3. No dragon caves can be found.

**Expected behavior**
I expected the dragon caves to still generate into a new world as long as I keep it = true in the config.

**Comment by wilyank** [comment]
I also experienced this issue. Further information:
upon loading the world, when it prints all the generation config to the console, it gives this.
`{SPAWN_DRAGON_SKELETON_I_CF=true, HYDRA_CAVE_CF=true, SPAWN_SEA_SERPENT_CF=true, SPAWN_STYMPHALIAN_BIRD_CF=true, SPAWN_WANDERING_CYCLOPS_CF=true, PIXIE_VILLAGE_CF=false, SPAWN_DEATH_WORM_CF=true, FIRE_DRAGON_CAVE_CF=false, SIREN_ISLAND_CF=true, MYRMEX_HIVE_DESERT_CF=true, SPAWN_HIPPOCAMPUS_CF=true, FIRE_DRAGON_ROOST_CF=false, MYRMEX_HIVE_JUNGLE_CF=true, GORGON_TEMPLE_CF=true, COPPER_ORE_CF=true, SILVER_ORE_CF=true, LIGHTNING_DRAGON_CAVE_CF=false, SPAWN_DRAGON_SKELETON_F_CF=true, AMETHYST_ORE_CF=true, SPAWN_DRAGON_SKELETON_L_CF=true, LIGHTNING_LILY_CF=true, MAUSOLEUM_CF=true, ICE_DRAGON_CAVE_CF=false, GRAVEYARD_CF=true, SAPPHIRE_ORE_CF=true, LIGHTNING_DRAGON_ROOST_CF=false, CYCLOPS_CAVE_CF=true, FROST_LILY_CF=true, FIRE_LILY_CF=true, ICE_DRAGON_ROOST_CF=false}`
While in the config, I set 
`"Generate Dragon Roosts" = false` 
and
`"Generate Dragon Caves" = true`

**Comment by Dejeh** [comment]
I can confirm this too, literally just encountered this problem. Turning dragon caves to false (and roosts to true) does not oppose this problem, it works as it should. The only problem is that turning roosts to false disables dragons entirely.
